adapters/repos/db/replication.go

```diff
@@ -319,3 +320,32 @@ func (s *Shard) reinit(ctx context.Context) error {
 
 	return nil
 }
+
+func (db *DB) OverwriteObjects(ctx context.Context,
+	objects []*objects.VObject,
+) ([]int64, error) {
+	result := make([]int64, len(objects))
+	for i, vobj := range objects {
+		found, err := db.Object(ctx,
+			vobj.Object.Class().String(), vobj.Object.ID(),
+			nil, additional.Properties{}, nil)
+		if err != nil {
+			return nil, err
+		}
+		// the db's stored object is not the most recent version.
+		// in this case, we overwrite it with the more recent one
+		if found.Object().LastUpdateTimeUnix < vobj.Version {
+			// 1) overwrite object
+			err := db.PutObject(ctx, &vobj.Object.Object, vobj.Object.Vector)
```

Comment:
db.PutObject() is calling index.putObject() under the hood which might call the coordinator.
We need to make sure that this function only locally executed (i.e. on local shards)

Reply:
done